### PR TITLE
[SSPROD-48724] CFT template for VM Workload Scanning

### PR DESCRIPTION
**Pull request by miguelpais**

### CODE REVIEW

**Review by cgeers — CHANGES_REQUESTED** [review]
The logical permutations of org and lambda enabled seem duplicative and unnecessarily brittle.

Instead, I'd advocate using an [IAM::RolePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-rolepolicy.html) to conditionally apply polices when needed, rather than duplicating entire resource blocks.

**Comment by miguelpais** [comment]
> The logical permutations of org and lambda enabled seem duplicative and necessarily brittle.
> 
> Instead, I'd advocate using an [IAM::RolePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-rolepolicy.html) to conditionally apply polices when needed, rather than duplicating entire resource blocks.



> The logical permutations of org and lambda enabled seem duplicative and unnecessarily brittle.
> 
> Instead, I'd advocate using an [IAM::RolePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-rolepolicy.html) to conditionally apply polices when needed, rather than duplicating entire resource blocks.

Handled, please re-check.

**Review by cgeers — APPROVED** [review]
looks great

File: modules/vm_workload_scanning.cft.yaml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                      Action:
+                        - ecr:GetDownloadUrlForLayer
+                        - ecr:BatchGetImage
+                        - ecr:BatchCheckLayerAvailability
+                        - ecr:ListImages
+                        - ecr:GetAuthorizationToken
+                      Resource: '*' 
+            LambdaPolicy:
+              Type: AWS::IAM::Policy
+              Condition: IsOrganizationalAndLambdaEnabled
```
Comment (ravinadhruve10) [inline modules/vm_workload_scanning.cft.yaml:232]: @miguelpais we don't have the `IsOrganizationalAndLambdaEnabled` conditional defined anywhere anymore. so this should be `IsLambdaEnabled` right?

Comment (miguelpais) [reply modules/vm_workload_scanning.cft.yaml]: thank you so much! Addressed!

**Review by ravinadhruve10 — APPROVED** [review]
Looks good! Thanks @miguelpais !

**Pull request merged**